src/FakeItEasy/Core/SequentialCallContext.cs

```diff
@@ -2,48 +2,51 @@ namespace FakeItEasy.Core
 {
     using System;
     using System.Collections.Generic;
-    using System.Globalization;
-    using System.IO;
+    using System.Linq;
 
-    internal class OrderedFakeAsserter : IFakeAsserter
+    internal class SequentialCallContext : ISequentialCallContext
     {
-        private readonly IEnumerable<IFakeObjectCall> originalCallList;
         private readonly CallWriter callWriter;
-        private readonly Queue<IFakeObjectCall> calls;
-        private readonly List<AssertedCall> assertedCalls = new List<AssertedCall>();
+        private readonly HashSet<FakeManager> fakeManagers;
+        private readonly List<AssertedCall> assertedCalls;
+        private int currentSequenceNumber;
 
-        public OrderedFakeAsserter(IEnumerable<IFakeObjectCall> calls, CallWriter callWriter)
+        public SequentialCallContext(CallWriter callWriter)
         {
-            Guard.AgainstNull(calls, "calls");
-            Guard.AgainstNull(callWriter, "callWriter");

```

Comment:
Yes, probably.

Actually, the diff did something funny here. It made it look as if `SequentialCallContext` was a rename and modification of `OrderedFakeAsserter`, but the truth is that it was written as an entirely new class, and I later removed `OrderedFakeAsserter`. So I never actually removed the guard clause, I just didn't put it ;)

I'll put it back anyway.